upgrade/upgrade-scripts/rootfs-container

```diff
@@ -19,14 +19,6 @@
 
 CONTAINER=
 
-function get_dataset_snapshots() {
-	zfs list -rt snapshot -Hpo name "$1"
-}
-
-function get_snapshot_clones() {
-	zfs get clones -Hpo value "$1"
-}
-
 function delete() {
```

Comment:
Interestingly, I don't think this function is ever used anywhere?

Reply:
Unless I'm misunderstanding something, it's used when you execute `rootfs-container delete ...`; see [here](https://github.com/delphix/appliance-build/blob/master/upgrade/upgrade-scripts/rootfs-container#L260)

Reply:
Right, but I can't find any users of `rootfs-container delete` ?

Reply:
Ah yes.. so, right now we don't really have a "dx_delete" equivalent for our linux appliance, but I wrote this "delete" function with the assumption that it could/would be used when we get around to implementing "dx_delete" for linux.

My assumption being, at some point we'll want to use "not-in-place" upgrades via the app-gate upgrade logic, so we'll eventually need a way to delete the "old version/container", and that's where this comes into play.

In the mean time, it can be useful during development, as a way to "manually" remove old containers when doing testing that uses "not-in-place" (e.g. I've used it when verifying this change, to clean up after the tests that I run).